tests/basic/determinism_check.rs

```diff
@@ -0,0 +1,94 @@
+// use crate::check_replay_roundtrip;
+use shuttle::rand::{thread_rng, Rng};
+// use shuttle::scheduler::RandomScheduler;
+use shuttle::sync::{Mutex, Arc};
+use shuttle::{check_determinism, thread};
+use test_log::test;
+
+
+#[test]
+#[should_panic]
+fn shuttle_determinism_test1() {
+    check_determinism(|| {
+
+        const NUM_THREADS: u32 = 10;
+
+        let lock = Arc::new(Mutex::new(0u64));
+        let threads: Vec<_> = (0..NUM_THREADS)
+        .map(|_| {
+
+            let my_lock = lock.clone();
+
+            thread::spawn(move || {
+                let x = thread_rng().gen::<u64>();
+
+				// Fail every x threads
+                if x % 1000 == 0 {
```

Comment:
Is this test going to be reliable enough to fail all the time? Wonder if we should just make this `x % 10` or something so it's always going to fail.